gpytorch/utils/cholesky.py

```diff
@@ -10,19 +10,21 @@
 def batch_potrf(mat):
     """
     """
-    potrf_list = []
-    for i in range(mat.size(0)):
-        potrf_list.append(mat[i].potrf().unsqueeze(0))
-    return torch.cat(potrf_list, 0)
+    potrf_list = [sub_mat.potrf() for sub_mat in mat.view(-1, *mat.shape[-2:])]
+    res = torch.cat(potrf_list, 0)
+    return res.view_as(mat)
 
 
 def batch_potrs(mat, chol):
```

Comment:
Can you add a TODO here for using batch cholesky decomp / solve once https://github.com/pytorch/pytorch/pull/11796 is done?

Reply:
(same up for potrf of course)

Reply:
woohoo! Any rough estimates on how much faster this is compared to looping through?

Reply:
I am seeing at least a 2x speed improvement when compared with the implementation here. These are CPU times though:

| mat_dim | chol_dim | looping time | native_op_time |
| ----------- | ------------ | --------------- | ------------------- |
| 1000 x 3 x 3 | 1000 x 3 x 3 | 11.9 ms ± 53.8 µs per loop | 667 µs ± 10.7 µs per loop |
| 1000 x 5 x 5 | 1000 x 5 x 5 | 12.3 ms ± 264 µs per loop | 1.32 ms ± 84.4 µs per loop |
| 1000 x 20 x 20 | 1000 x 20 x 20 | 19.5 ms ± 162 µs per loop | 7.61 ms ± 501 µs per loop |
| 1000 x 50 x 50 | 1000 x 50 x 50 | 42.6 ms ± 1.96 ms per loop | 21.2 ms ± 1.87 ms per loop |